konvoy: Update docs for konvoy release v1.5.0-rc.1 (#3055)
mesosphere/dcos-docs-site: pages/ksphere/konvoy/1.5/upgrade/upgrade-cli/index.md

---
layout: layout.pug
navigationTitle: Upgrade Konvoy CLI
title: Upgrade Konvoy CLI
menuWeight: 30
excerpt: Upgrade Konvoy image CLI version
enterprise: false
---

<p class="message--note"><strong>NOTE: </strong>Before upgrading, keep in mind that your Konvoy CLI version can only be upgraded to a compatible Konvoy version.</p>

## Konvoy CLI available versions

> You can verify to which version your CLI can be upgraded by running the following command:

Reviewer: ```suggestion
You can verify which version your CLI can be upgraded to by running the following command:
```

```bash
konvoy image list
  KONVOY VERSION | DEFAULT KUBERNETES VERSION
+----------------+----------------------------+
  v1.1.4         | 1.15.3
```

This command lists all the available versions to which your current CLI can be upgraded.
This list also shows the default Kubernetes version of each Konvoy version.

This command uses the Docker Hub to fetch all the available Konvoy versions.

<p class="message--note"><strong>NOTE: </strong>This functionality is only supported for CLI where the Konvoy version is greater or equal to <strong>v1.1.4</strong>.</p>

### Using a private Docker registry

If you are using a <a href="https://docs.docker.com/registry/deploying/" target="_blank">private Docker registry</a> for your clusters, you can list all the available versions, passing some additional arguments to the previous command.

> For instance, if your private Docker registry provides a basic authentication mechanism (username/password), you need to pass the following arguments, where `docker-registry-skip-verify` is optional based on your TLS settings:

Reviewer: ```suggestion
For example, if your private Docker registry provides a basic authentication mechanism (username/password), you must pass the following arguments, where `docker-registry-skip-verify` is optional based on your TLS settings:
```

```bash
konvoy image list --docker-registry-url=https://localhost:6443 --docker-registry-username=testuser --docker-registry-password=testpassword --docker-registry-skip-verify
```

If you are using a Docker registry with <a href="https://docs.docker.com/registry/spec/auth/token/" target="_blank">v2 token authentication mechanism</a> enabled, then you should set the following arguments to be able to list the CLI versions from the Docker registry API.

> For instance, if you are running <a href="https://github.com/goharbor/harbor" target="blank">Harbor</a> Docker Registry, `konvoy image` talks to the <a href="https://github.com/docker/distribution/blob/master/docs/spec/auth/token.md" target="_blank">API to authenticate</a> to get some data about the available versions.

Reviewer: ```suggestion
For example, if you are running <a href="https://github.com/goharbor/harbor" target="blank">Harbor</a> Docker Registry, `konvoy image` talks to the <a href="https://github.com/docker/distribution/blob/master/docs/spec/auth/token.md" target="_blank">API to authenticate</a> to get some data about the available versions.
```

```bash
konvoy image list --docker-registry-url=https://myregistry.com --docker-registry-username=admin --docker-registry-password=Harbor12345  --docker-registry-skip-verify
```

If you pulled and pushed the Konvoy Docker image under a different image name or Docker repository in your registry, you need to include an additional argument to the command `docker-registry-repository`.

As an example, we pushed the Konvoy image under the `library` public repository of our Docker repository `--docker-registry-repository=library/mesosphere/konvoy`:

```bash
konvoy image list --docker-registry-url=https://localhost:6443 --docker-registry-username=testuser --docker-registry-password=testpassword --docker-registry-skip-verify --docker-registry-repository=library/mesosphere/konvoy
```

## Konvoy CLI version upgrade

> Once you listed the available Konvoy versions, you can upgrade your CLI by running the following command:

Reviewer: ```suggestion
After you have the available Konvoy versions, you can upgrade your CLI by running the following command:
```

```bash
konvoy image upgrade --version=v1.1.4
Wrote Konvoy CLI version 'v1.1.4' to '.konvoy/cli_version'
```

> Once the upgrade command completed, you can start using the new Konvoy version.

Reviewer: ```suggestion
After the upgrade command completes, you can start using the new Konvoy version.
```

## Before you begin: Prepare for Konvoy CLI upgrade

If you're using a newer version of the Konvoy CLI, it may require changes to your `cluster.yaml`.
These changes are highlighted in the Release Notes and in the following sections.

A Konvoy upgrade consists of a few distinct steps.

- Download the Konvoy binary and extract it in your environment in the same manner as the initial install. (Download the specific version of Konvoy by following the steps outlined in the [download](../../download) page.)

> - You must also have access to the `cluster.yaml` file and the SSH keys that were generated during the initial install.. Update the `cluster.yaml` file with the changes outlined below.

Reviewer: ```suggestion
- You must also have access to the `cluster.yaml` file and the SSH keys that were generated during the initial install. Update the `cluster.yaml` file with the changes described below.
```
